src/hotspot/cpu/x86/x86.ad

```diff
@@ -1935,6 +1944,22 @@ bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType bt) {
         return false;
       }
       break;
+    case Op_SaturatingAddVB:
+    case Op_SaturatingAddVS:
+    case Op_SaturatingSubVB:
+    case Op_SaturatingSubVS:
+       if (UseAVX < 1 || size_in_bits < 128 || (size_in_bits == 512 && !VM_Version::supports_avx512bw())) {
+         return false;
+       }
```

Comment:
UseAVX < 1 could be written as UseAVX == 0. Could we not do register version for size_in_bit < 128?

Reply:
I get your point, but constraints ensure we only address cases with vector size >= 128 bit in this patch..